phoenix-core/src/main/java/org/apache/phoenix/util/ViewUtil.java

```diff
@@ -1090,4 +1099,202 @@ public static void addTagsToPutsForViewAlteredProperties(List<Mutation> tableMet
+                                new TableInfo(tenantId.getBytes(), schemCol.getBytes(),
+                                        tName.getBytes());
+                        tenantViewSet.add(tableInfo);
+                    }
+                }
+            }
+        }
+        return tenantViewSet;
+    }
+
+    private static void getTTLInfo(Set<TableInfo> viewSet, Configuration configuration,
+            boolean isIndexTable, List<TableTTLInfo> tableTTLInfoList)
+            throws SQLException {
+
+        if (viewSet.size() == 0) {
+            return;
+        }
+        String
+                viewsClause =
+                new StringBuilder(viewSet.stream()
+                        .map((v) -> String.format("(%s, %s,'%s')",
+                                Bytes.toString(v.getTenantId()),
+                                Bytes.toString(v.getSchemaName()),
+                                Bytes.toString(v.getTableName())))
+                        .collect(Collectors.joining(","))).toString();
+        String
+                viewsWithTTLSQL =
+                "SELECT TENANT_ID, TABLE_SCHEM, TABLE_NAME, " +
+                        "TABLE_TYPE, TTL, ROW_KEY_MATCHER " +
+                        "FROM SYSTEM.CATALOG " +
+                        "WHERE TABLE_TYPE = 'v' AND " +
+                        "(TENANT_ID, TABLE_SCHEM, TABLE_NAME) IN " +
+                        "(" + viewsClause.toString() + ")";
+        logger.debug(
+                String.format("ViewsWithTTLSQL : %s", viewsWithTTLSQL));
+
+        try (Connection serverConnection = QueryUtil.getConnectionOnServer(new Properties(),
+                configuration)) {
+
+            try (
+                    PhoenixPreparedStatement
+                            viewTTLStmt =
+                            serverConnection.prepareStatement(viewsWithTTLSQL)
+                                    .unwrap(PhoenixPreparedStatement.class)) {
+
+                try (ResultSet viewTTLRS = viewTTLStmt.executeQuery()) {
+                    while (viewTTLRS.next()) {
+                        String tid = viewTTLRS.getString("TENANT_ID");
+                        String schem = viewTTLRS.getString("TABLE_SCHEM");
+                        String tName = viewTTLRS.getString("TABLE_NAME");
+                        byte[]
+                                tenantIdBytes =
+                                tid == null || tid.isEmpty() ? EMPTY_BYTE_ARRAY : tid.getBytes();
+
+                        String fullTableName = SchemaUtil.getTableName(schem, tName);
+                        Properties tenantProps = new Properties();
+                        if (tid != null) {
+                            tenantProps.setProperty(PhoenixRuntime.TENANT_ID_ATTRIB, tid);
+                        }
+                        Connection
+                                tableConnection =
+                                QueryUtil.getConnectionOnServer(tenantProps, configuration);
```

Comment:
The connection is not closed here. I suggest opening it within a try statement. Also, can we use one global connection instead of a separate connection for every view?